app/models/repository.rb

```diff
@@ -6,14 +8,16 @@ class Repository < ApplicationRecord
 
   attribute :discarded_by_id, :integer
 
-  belongs_to :team, optional: true
+  belongs_to :team
   belongs_to :created_by, foreign_key: :created_by_id, class_name: 'User'
   has_many :repository_columns, dependent: :destroy
   has_many :repository_rows, dependent: :destroy
   has_many :repository_table_states,
            inverse_of: :repository, dependent: :destroy
   has_many :report_elements, inverse_of: :repository, dependent: :destroy
   has_many :repository_list_items, inverse_of: :repository, dependent: :destroy
+  has_many :team_repositories, inverse_of: :repository, dependent: :destroy
+  has_many :teams_shared_with, through: :team_repositories, class_name: 'Team'
```

Comment:
shared_with_teams maybe?

Reply:
Wil stays `has_many :teams_...` for now...